src/highlight/HighlightAnnotations.tsx

```diff
@@ -1,46 +1,137 @@
+    staged?: CreatorHighlight | null;
+    status: CreatorStatus;
 };
 
-export default class HighlightAnnotations extends React.PureComponent<Props> {
-    static defaultProps = {
-        annotations: [],
-        isCreating: false,
-    };
+const HighlightAnnotations = (props: Props): JSX.Element => {
+    const {
+        activeAnnotationId,
+        annotations = [],
+        createHighlight = noop,
+        isCreating = false,
+        location,
+        message,
+        setActiveAnnotationId,
+        setMessage,
+        setMode,
+        setStaged,
+        setStatus,
+        staged,
+        status,
+    } = props;
+    const [highlightRef, setHighlightRef] = React.useState<HTMLAnchorElement | null>(null);
 
-    handleAnnotationActive = (annotationId: string | null): void => {
-        const { setActiveAnnotationId } = this.props;
+    const canReply = status !== CreatorStatus.started && status !== CreatorStatus.init;
+    const isPending = status === CreatorStatus.pending;
 
+    const handleAnnotationActive = (annotationId: string | null): void => {
         setActiveAnnotationId(annotationId);
     };
 
-    render(): JSX.Element {
-        const { activeAnnotationId, annotations, isCreating } = this.props;
-        const sortedAnnotations = annotations.filter(isValidHighlight).sort(sortHighlight);
+    const handleCancel = (): void => {
+        setMessage('');
+        setStaged(null);
+        setStatus(CreatorStatus.init);
+    };
 
-        return (
-            <>
-                {/* Layer 1: Saved annotations -- visual highlights */}
-                <HighlightCanvas activeId={activeAnnotationId} annotations={sortedAnnotations} />
+    const handleChange = (text?: string): void => {
+        setMessage(text || '');
+    };
 
-                {/* Layer 2: Saved annotations -- interactable highlights */}
-                <HighlightList annotations={sortedAnnotations} onSelect={this.handleAnnotationActive} />
+    // TODO: For testing purposes only
+    const handleClick = (): void => {
+        // Page 1 "troublesome personality" of the Van Gogh doc
+        const shapes = [
+            {
+                height: 2.0239190432,
+                width: 21.1208338959,
+                x: 38.7415889335,
+                y: 75.0991835327,
+                type: 'rect' as const,
+            },
+        ];
 
-                {/* Layer 3: Drawn (unsaved) incomplete annotation target, if any */}
-                {isCreating && <HighlightCreator className="ba-HighlightAnnotations-creator" />}
-            </>
-        );
-    }
-}
+        setMode(Mode.HIGHLIGHT);
+        setStatus(CreatorStatus.staged);
+        setStaged({ target: { location: { value: location, type: 'page' }, shapes, type: 'highlight' } });
+    };
+
+    const handleSubmit = (): void => {
+        if (!staged) {
+            return;
+        }
+
+        createHighlight({ ...staged, message });
+    };
+
+    return (
+        <>
+            <button
```

Comment:
This layer is not currently rendered by default, correct? Otherwise existing users will see a random button.

Reply:
Yeah I'll be removing this testing code before merging in this PR